storm-server/src/main/java/org/apache/storm/DaemonConfig.java

```diff
@@ -792,6 +792,12 @@
     public static final String ISOLATION_SCHEDULER_MACHINES = "isolation.scheduler.machines";
 
     /**
+     * Configuration elements for scheduler config loader.
+     */
+    @isMapEntryType(keyType = String.class, valueType = String.class)
```

Comment:
So I made a mistake.  I thought initially that each of the configs in `ConfigLoaderConfiguration` were regular configs under storm.yaml.  I should have read the code more closely, because they actually are sub-configs under "scheduler.config.loader.params".

I don't really like them being sub configs because they don't get validated.  Which was the entire point of me asking you to add them into the Validated ServiceLoader.  I also don't see a lot of value in having

```yaml
scheduler.config.loader.params:
    scheduler.config.loader.uri: "artifactory+http://artifactory.my.company.com:8000/artifactory/configurations/clusters/my_cluster/ras_pools"
    scheduler.config.loader.timeout.sec: 30
```

over 

```yaml
scheduler.config.loader.uri: "artifactory+http://artifactory.my.company.com:8000/artifactory/configurations/clusters/my_cluster/ras_pools"
scheduler.config.loader.timeout.sec: 30
```

Could we change it?

Reply:
You are right. I also felt that `scheduler.config.loader.params` makes things unnecessarily complicated. Changing it